### PR TITLE
Partial preemption for groups with multiple sequences

### PR DESCRIPTION
**Pull request by popovaan**
- Partial preemption for groups with multiple sequences.
- Fixed a bug in can_append_stots()

Ticket: 140648

### CODE REVIEW

**Review by iefode — APPROVED** [review]

**Comment by iefode** [comment]
LGTM

**Review by pavel-esir — APPROVED** [review]

File: src/cpp/continuous_batching/src/scheduler.hpp [hunk 1]
```diff
@@ -101,52 +101,45 @@ class Scheduler {
...
-            else {
-                m_block_manager.free_sequence_partially(seq_id, required_blocks);
-            }
-
-            // calculate the number of released blocks
-            auto released_blocks = m_block_manager.num_free_blocks() - prev_blocks_count;
-            total_num_released_blocks += released_blocks;
-            prev_blocks_count = m_block_manager.num_free_blocks();
-
+        if (num_running_sequences > 1) {
```
[the remark this answers is not in the document]
Comment (popovaan) [reply src/cpp/continuous_batching/src/scheduler.hpp:117]: Removed distinguishing between these cases in [PR](https://github.com/openvinotoolkit/openvino.genai/pull/730) as discussed.

File: src/cpp/continuous_batching/src/block_manager.hpp [hunk 2]
```diff
@@ -110,6 +110,78 @@ class BlockManager {
...
+                blocks_released++;
+            }
+        }
+        return blocks_released;
+    }
+
+    const bool free_group_partially_multiple_runnning_sequence(SequenceGroup::Ptr sequence_group, size_t num_required_blocks, size_t& phisical_blocks_released, size_t& logical_blocks_released) {
+        phisical_blocks_released = 0;
+        logical_blocks_released = 0;
+        while (num_required_blocks > phisical_blocks_released) {
```
[the remark this answers is not in the document]
Comment (popovaan) [reply src/cpp/continuous_batching/src/block_manager.hpp:131]: Simplified this method using formula `blocks_to_remove_per_sequence == ceil(num_required_blocks / sequence_num)`.

File: src/cpp/continuous_batching/src/block_manager.hpp [hunk 3]
```diff
@@ -110,6 +110,78 @@ class BlockManager {
         return m_block_table[seq_id];
     }
 
+    const size_t free_rightest_blocks(SequenceGroup::Ptr sequence_group) {
```
[the remark this answers is not in the document]
Comment (popovaan) [reply src/cpp/continuous_batching/src/block_manager.hpp:113]: Removed this method.

**Pull request merged**